doc/source/building/compilers_and_options.rst

```diff
@@ -0,0 +1,118 @@
+Compiler selection and customizing a build
+==========================================
+
+Selecting a specific compiler
+-----------------------------
+
+Meson supports the standard environment variables ``CC``, ``CXX`` and ``FC`` to
+select specific C, C++ and/or Fortran compilers. These environment variables are
+documented in `the reference tables in the Meson docs
+<https://mesonbuild.com/Reference-tables.html#compiler-and-linker-flag-environment-variables>`__.
+
+Note that environment variables only get applied from a clean build, because
+they affect the configure stage (i.e., ``meson setup``). An incremental rebuild
+does not react to changes in environment variables - you have to run ``git
+clean -xdf`` and do a full rebuild, or run ``meson setup --reconfigure``.
+
+
+Adding a custom compiler or linker flag
+---------------------------------------
+
+Meson by design prefers builds being configured through command-line options
+passed to ``meson setup``. It provides many built-in options:
+
+- For enabling a debug build and the optimization level, see the next section
+  on "build types",
+- Enabling ``-Werror`` in a portable manner is done via ``-Dwerror=true``,
+- Enabling warning levels is done via ``-Dwarning_level=<int>``,
```

Comment:
There is, technically, a warning_level=everything option, which provides a portable way to do clang -Weverything, MSVC /Wall, and GCC -Wlong -Wlist -Wof -Woptions (because GCC doesn't want to provide a convenient option to enable this).